File: 卡码/代码随想录/1/06-二叉树/C++/01-二叉树遍历方式/LeetCode_0102.cpp
//102. 二叉树的层序遍历
#include <iostream>
#include <vector>
#include <queue>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
	vector<vector<int>> levelOrder(TreeNode* root) {
		vector<vector<int>> res;
		vector<int> v;
		queue<TreeNode*> q;
		if(root!=nullptr) q.push(root);
		int size;
		while(!q.empty()){
			size=q.size();	//当前层元素个数
			v.clear();	//清空数组
			while(size-->0){
				TreeNode* node=q.front();
				q.pop();
				v.push_back(node->val);
				if(node->left) q.push(node->left);
				if(node->right) q.push(node->right);
			}
			res.push_back(v);
		}
		return res;
	}
};

int main(){
	system("pause");
	return 0;
}

File: 卡码/代码随想录/1/06-二叉树/Js/01-二叉树遍历方式/LeetCode_0102.js
//102. 二叉树的层序遍历
import TreeNode from "../TreeNode.js";

const levelOrder = function(root) {
  const res=new Array();
  const queue=new Array();
  if(root) queue.push(root);
  while(queue.length>0){
    let size=queue.length;
    const v=new Array();
    while(size-->0){
      let node=queue.shift();
      v.push(node.val);
      if(node.left) queue.push(node.left);
      if(node.right) queue.push(node.right);
    }
    res.push(v);
  }
  return res;
};